File: ios/Runner/Feature/Pose/PoseError.swift
//
//  PoseError.swift
//  Runner
//
//  Created by 吉田周平 on 2022/11/26.
//

import Foundation

enum PoseError: Int, NativeError {
    
    case detect
    case imageNotFound
    case unexpected
    
    var message: String {
        switch self {
        case .detect:
            return "During detecting pose occured error"
        case .imageNotFound:
            return "UIImage not found"
        case .unexpected:
            return "Unexpected Error"
        }
    }
    
    var isSuccess: Bool { return false }
    
    func toDict() -> [String : Any?] {
        return [
            "result": isSuccess,
            "code": self.rawValue,
            "message": message,
        ]
    }
    
}

File: android/app/src/main/kotlin/com/example/shus_flutter/feature/pose/PoseError.kt
package com.example.shus_flutter.feature.pose

import com.example.shus_flutter.common.`interface`.NativeError
import java.util.*

enum class PoseError(override val code: Int, override val message: String) : NativeError {

    DETECT(0, "During detecting pose occurred error"),
    IMAGENOTFOUND(1, "UIImage not found"),
    UNEXPEXTED(2, "Unexpected Error");

    override fun toData(): HashMap<String, Any> {
        var result: HashMap<String, Any> = HashMap<String, Any>()
        result.put("result", isSuccess)
        result.put("code", code)
        result.put("message", message)
        return  result
    }
}